File: 30-Days-Of-Code/Day 10/binary.cpp
#include <iostream>

using namespace std;
int main() {
    int n;
    cin >> n;

    int maxStreak = 0;
    int currStreak = 0;

    while (n) {
        if (n % 2) {
            currStreak++;
            if (currStreak > maxStreak) {
                maxStreak = currStreak;
            }
        } else {
            currStreak = 0;
        }
        n >>= 1;
    }

    cout << maxStreak << endl;
    return 0;
}

File: Learn/30-Days-Of-Code/Day 10/binary.py
n = int(input().strip())

curr_streak = 0
max_streak = 0
while n > 0:
    if n % 2 == 1:
        curr_streak += 1
        if curr_streak > max_streak:
            max_streak = curr_streak
    else:
        curr_streak = 0
    n //= 2

print(max_streak)

